### PR TITLE
Make params exclusive to `COp`s

### LINKED ISSUE #511: Remove params in python perform method
### Description

`params` are used to write C code that is shared among different parametrizations of the same Op class, such as an Op with `self.inplace = True` vs `False`: https://pytensor.readthedocs.io/en/latest/extending/using_params.html

They are also supposed to be passed to the python `perform` method (for those Ops that implement it), which we forgot to do when we implemented `Blockwise`. This seems unnecessary, and in fact many Ops just ignore params, and use node properties directly:
https://github.com/pymc-devs/pytensor/blob/9cf2d181f07dc99bbd2e7c9e2b4a3e1b0aeff034/pytensor/tensor/basic.py#L3998-L4002

I could only find 3 cases where it was being used, including:
https://github.com/pymc-devs/pytensor/blob/9cf2d181f07dc99bbd2e7c9e2b4a3e1b0aeff034/pytensor/tensor/blas.py#L920-L921

The complication of the signature doesn't seem justified, for something that is clearly focused on the C backend. There is even a warning in the docs linked above:

image

### PR DESCRIPTION
**Pull request by ricardoV94**
Closes #511

### CODE REVIEW

File: pytensor/link/c/basic.py [hunk 1]
```diff
@@ -803,7 +809,10 @@ def code_gen(self):
 
             sub = dict(failure_var=failure_var)
 
-            params = node.run_params()
+            try:
```
Comment (ricardoV94) [inline pytensor/link/c/basic.py:812]: We raise 4 lines above if it's not a ClinkerOp

**Comment by ricardoV94** [comment]
Would like to get this merged before the next release

**Review by twiecki — APPROVED** [review]
Nice simplification. I guess the COps still need params?

**Comment by ricardoV94** [comment]
Yup

**Pull request merged**